### PR TITLE
[HER-96] Change detections repository

### PR DESCRIPTION
**Pull request by ConstantinescuAndrei**
## Description
- add filter to the fetch method
- add Camera field to entity
- change detection factory

## Related
[HER-96](https://nerds-interns.atlassian.net/jira/software/projects/HER/boards/1?issueParent=10022&selectedIssue=HER-96)

### CODE REVIEW

File: src/Presentation/Detection/Fetching/Resolver.cs [hunk 1]
```diff
@@ -11,7 +13,11 @@ public class Resolver : IResolver
 
         public Resolver(IGetDetection getDetection) => _getDetection = getDetection;
 
-        public IEnumerable<MDetection> Execute(IResolveFieldContext<object> input) =>
-            _getDetection.Execute(new Nothing());
+        public IEnumerable<MDetection> Execute(IResolveFieldContext<object> input)
+        {
+            var detections = _getDetection.Execute(new PDetection());
```
Comment (danidoff) [inline src/Presentation/Detection/Fetching/Resolver.cs:18]: Should we remove this variable and return the command here ?

**Review by danidoff — CHANGES_REQUESTED** [review]

**Review by RaduBuciuceanu — DISMISSED** [review]

File: src/Presentation/Detection/Fetching/Resolver.cs (hunk 1 above)
Comment (ConstantinescuAndrei) [reply src/Presentation/Detection/Fetching/Resolver.cs]: done

File: src/Data/Detection/Filtering/DetectionFilter.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        }
+
+        private IQueryable<EDetection> MatchId(IQueryable<EDetection> input) =>
+            _parameter.Id != null ? input.Where(detection => detection.Id == _parameter.Id) : input;
+
+        private IQueryable<EDetection> MatchScore(IQueryable<EDetection> input) =>
+            _parameter.Score != null ? input.Where(detection => detection.Score == _parameter.Score) : input;
+
+        private IQueryable<EDetection> MatchClass(IQueryable<EDetection> input) =>
+            string.IsNullOrEmpty(_parameter.Class) ? input
```
Comment (danidoff) [inline src/Data/Detection/Filtering/DetectionFilter.cs:34]: Should we move to another line the `? input` ?

**Review by danidoff — CHANGES_REQUESTED** [review]

File: src/Data/Detection/Filtering/DetectionFilter.cs (hunk 2 above)
Comment (ConstantinescuAndrei) [reply src/Data/Detection/Filtering/DetectionFilter.cs]: done

**Review by danidoff — APPROVED** [review]

File: src/Business/Detection/Fetching/Models/PDetection.cs [hunk 3]
```diff
@@ -5,9 +5,9 @@ namespace Business.Detection.Fetching.Models
 
     public class PDetection
     {
-        public int Id { get; set; }
+        public int? Id { get; set; }
 
-        public int Score { get; set; }
+        public int? Score { get; set; }
```
Comment (mihairadulescu) [inline src/Business/Detection/Fetching/Models/PDetection.cs:10]: why nullable int ?

**Review by mihairadulescu — CHANGES_REQUESTED** [review]
just leave that int not nullale, with default value of 0... that's all.

File: src/Business/Detection/Fetching/Models/PDetection.cs (hunk 3 above)
Comment (RaduBuciuceanu) [reply src/Business/Detection/Fetching/Models/PDetection.cs:10]: - null = don't filter
- value = filter by value

**Review by RaduBuciuceanu — APPROVED** [review]

**Review by mihairadulescu — APPROVED** [review]

**Pull request merged**